regression/synthetic_dataset.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2023 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Create a toy dataset using MDSWriter for regression testing."""
+
+import os
+import shutil
+import tempfile
+from argparse import ArgumentParser, Namespace
+from typing import Union
+
+import numpy as np
+
+from streaming import MDSWriter
+
+_NUM_SAMPLES = 10000
+# Word representation of a number
+_ONES = ('zero one two three four five six seven eight nine ten eleven twelve '
+         'thirteen fourteen fifteen sixteen seventeen eighteen nineteen').split()
+_TENS = 'twenty thirty forty fifty sixty seventy eighty ninety'.split()
+
+_COLUMNS = {
+    'number': 'int',
+    'words': 'str',
+}
+
+
+def parse_args() -> Namespace:
+    """Parse command-line arguments.
+
+    Returns:
+        Namespace: Command-line arguments.
+    """
+    args = ArgumentParser()
+    args.add_argument('--create', default=False, action='store_true')
```

Comment:
Is it possible to add one more parameter for the type of dataset that we need to create ? say right now we have `numberAndSay`, in future, we will have synthetic CV dataset !! See if you do with nested subparsers with argparse maybe ?

Reply:
I think we should keep this for now and when there's PR using CV it can charge the argument to specify a dataset type